utils/serverStatistic.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const { default: fetch } = require("node-fetch");
+
+const application_id = 'd78e7f67147305f3042bef05755fa168';
+const MAX_ACCOUNTS_PER_REQUEST = 100;
+const TOTAL_ACCOUNTS = 594859325;
+const BASE_URL = 'https://api.wotblitz.eu/wotb';
+const allPlayerStats = { account: {}, tanks: [] };
+
+const statsNames = ['battles', 'damage_dealt', 'damage_received', 'frags', 'spotted', 'wins', 'losses', 'survived_battles'];
+const achievementsName = ['markOfMastery', 'markOfMasteryI', 'markOfMasteryII', 'markOfMasteryIII'];
+const getStatsObject = (statsName, stats) => Object.fromEntries(statsName.map((statName) => [statName, stats[statName] || 0]));
+
+const getInfo = async (url) => {
+  try {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data.data;
+  } catch (error) {
+    console.error('Error:', error);
+  }
+};
+
+const getVehicleStats = async (accountId) => {
+  const result = {};
+  const vehicleStatsURL = `${BASE_URL}/tanks/stats/?application_id=${application_id}&account_id=${accountId}`;
+  const vehicleAchievmentsURL = `${BASE_URL}/tanks/achievements/?application_id=${application_id}&account_id=${accountId}`;
+  const vehicleStats = await getInfo(vehicleStatsURL);
+  const vehicleAchievments = await getInfo(vehicleAchievmentsURL);
+
+  const stats = vehicleStats[accountId];
+  const achievements = vehicleAchievments[accountId];
+
+  if (!stats || !achievements) return;
+
+  stats.map((tankStats) => {
+    const battleLifeTime = tankStats.battle_life_time;
+    if (!battleLifeTime) return;
+    const regular = getStatsObject(statsNames, tankStats.all);
+    result[tankStats.tank_id] = { regular, battleLifeTime };
+  });
+
+  achievements.map((tankAchivements) => {
+    if (!result[tankAchivements.tank_id]) return;
+    const mastery = getStatsObject(achievementsName, tankAchivements.achievements);
+    result[tankAchivements.tank_id].mastery = mastery;
+  });
+
+  return Object.entries(result).map(([wotId, data]) => ({ wotId, ...data }));
+};
```

Comment:
Simplified the logic for retrieving the mastery object using the find method to match the tank ID. The function now directly returns the mapped tank statistics array without the need for converting it using Object.entries.
```suggestion
  return stats.map((tankStats) => {
    const regular = getStatsObject(statsNames, tankStats.all);
    const mastery = getStatsObject(achievementsName, achievements.find((tankAchievements) => tankAchievements.tank_id === tankStats.tank_id).achievements);
    return { wotId: tankStats.tank_id, regular, battleLifeTime: tankStats.battle_life_time, mastery };
  });
```